DAQ/src/CaloHitsFromFragments_module.cc

```diff
@@ -63,94 +74,81 @@ struct CrystalInfo {
   // --- Production:
   virtual void produce(Event&);
 
-
 private:
   void analyze_calorimeter_(const artdaq::Fragment& f,
                             std::unique_ptr<mu2e::CaloHitCollection> const& calo_hits,
-                            std::unique_ptr<mu2e::CaloHitCollection> const& caphri_hits
-			    );
+                            std::unique_ptr<mu2e::CaloHitCollection> const& caphri_hits);
 
   void addPulse(uint16_t& crystalID, float& time, float& eDep);
 
   int diagLevel_;
 
   art::InputTag caloFragmentsTag_;
-  float         digiSampling_;
-  float         deltaTPulses_, pulseRatioMax_, pulseRatioMin_;
+  float digiSampling_;
+  float deltaTPulses_, pulseRatioMax_, pulseRatioMin_;
 
-  const int     hexShiftPrint = 7;
+  const int hexShiftPrint = 7;
 
-  std::array<std::vector<CrystalInfo>, 674*4>  pulseMap_;
-  mu2e::CaloDAQUtilities                       caloDAQUtil_;
+  std::unordered_map<uint16_t, std::list<CrystalInfo>>
+      pulseMap_; // Temporary hack until the Calorimeter channel map is finialized
```

Comment:
Is the choice of unordered map going to make results not repeatable if I run the same code 2 times on the same inputs?  For example if clustering code may be sensitive to the order of hits in a container.

Since this is a temporary hack repeatability is more important than speed.

Reply:
I think this can be replaced with a map without penalty